### PR TITLE
windows test support #275

### PR DESCRIPTION
**Pull request by arunavabasu-03**
## **Automate windows test** 
👉 python is not supported in travis;  so as a language I use `generic `and chocolaty to install python on the runner and the run tests, we can use references and anchors to gracefully compose our stages.

Hey @erwanp, please check that this approach is right or we have to go with another approach. 😀

### CODE REVIEW

**Comment by anandxkumar** [comment]
Hi @arunavabasu-03, great work. The Travis build has failed: https://app.travis-ci.com/github/radis/radis/jobs/563238622.
I would suggest you to build travis locally before pushing it to repo just to ensure it's not the syntax error or any other prominent issue causing it.
Reference: https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally

**Comment by arunavabasu-03** [comment]
thanks, @anandxkumar for your feedback 😄 .

**Comment by anandxkumar** [comment]
@erwanp seems like the Travis CI is not terminating, can you terminate it manually?

File: .travis.yml [hunk 1]
```diff
@@ -37,7 +36,7 @@ _test: &_test
     # Update python version in environment
     - sed -i -E 's/(python=)(.*)/\1'$TRAVIS_PYTHON_VERSION'/' ./environment.yml
     # Create conda environment
-    - conda env create -n radis-env -f environment.yml
+    - conda env create -n py36 radis-env -f environment.yml
```
Comment (erwanp) [inline .travis.yml:39]: default RADIS test environnment is now python 3.8, is there any specific reason you want to add py36?

Comment (arunavabasu-03) [reply .travis.yml:39]: hey @erwanp , **sorry for the late reply**. I just try with another YAML file. 😀

**Comment by arunavabasu-03** [comment]
hey @erwanp @anandxkumar , I add the micromamba support also.

**Comment by anandxkumar** [comment]
We will first have to solve the existing errors in https://github.com/radis/radis/pull/428, so test passes and then switch it by micromamba (https://github.com/radis/radis/pull/426) to check if it working as expected and then proceed with your PR to check if it's working with resolving any conflicts and check the added window test

**Comment by arunavabasu-03** [comment]
ok @anandxkumar, I am also trying to help you all  😃 .

File: .travis.yml (hunk 1 above)
Comment (erwanp) [reply .travis.yml]: I think the name should remain "radis-env" here

**Comment by erwanp** [comment]
@arunavabasu-03 we run RADIS on Travis open-source credits, which are limited. Also, we do not want to run too much calculations and do our part in saving 🌳  

Do you think it's possible to run the Windows tests only on pull-requests to the Master branch ? (that would be for every new versino, i.e. every ~2-3 months)

**Comment by anandxkumar** [comment]
Hi @arunavabasu-03, make sure to pull the latest updated develop branch, resolve any existing merge conflict before pushing the code next time!

File: .travis.yml [hunk 2]
```diff
@@ -15,53 +15,78 @@ stages:
...
+jobs:
+  include:
+    - <<: *_test
+      os: windows
+      language: shell
+      before_install:
+        - choco install python --version 3.9.2
+        - python --version
+        - choco install runinbash
+      env: PATH=/c/Python39:/c/Python39/Scripts:$PATH
```
Comment (anandxkumar) [inline .travis.yml:75]: @arunavabasu-03 shouldn't it be Python38?

File: .travis.yml [hunk 3]
```diff
@@ -15,53 +15,78 @@ stages:
...
+    - pip install codecov
+    - codecov
+
+jobs:
+  include:
+    - <<: *_test
+      os: windows
+      language: shell
+      before_install:
+        - choco install python --version 3.9.2
```
Comment (anandxkumar) [inline .travis.yml:72]: Shouldn't the python version for choco installation be the same?

File: .travis.yml (hunk 2 above)
Comment (anandxkumar) [inline .travis.yml:75]: It should get updated accordingly

**Comment by arunavabasu-03** [comment]
outdated !!

**Pull request closed without merging**